QgisModelBaker/libqgsprojectgen/dataobjects/layers.py

```diff
@@ -27,14 +27,15 @@
     QgsVectorLayer,
     QgsDataSourceUri,
     QgsWkbTypes,
-    QgsRectangle
+    QgsRectangle,
+    QgsCoordinateReferenceSystem
 )
-from qgis.PyQt.QtCore import QCoreApplication
+from qgis.PyQt.QtCore import QCoreApplication, QSettings
 
 
 class Layer(object):
 
-    def __init__(self, provider, uri, name, extent, geometry_column=None, wkb_type=QgsWkbTypes.Unknown, alias=None, is_domain=False, is_structure=False, is_nmrel=False, display_expression=None):
+    def __init__(self, provider, uri, name, srid, extent, geometry_column=None, wkb_type=QgsWkbTypes.Unknown, alias=None, is_domain=False, is_structure=False, is_nmrel=False, display_expression=None):
```

Comment:
a bit offtopic, but it would be nice to start using typehints:
```suggestion
    def __init__(self, provider: str, uri: str, name: str, srid: int, extent: QgsRectangle, geometry_column:str = None, wkb_type: QgsWkbTypes.Type = QgsWkbTypes.Unknown, alias:??? = None, is_domain:bool = False, is_structure:bool = False, is_nmrel:bool = False, display_expression:??? = None):
```